### ISSUE TITLE
Cannot cancel sign-in flow

### ISSUE THREAD

**Issue by bayo-code** [aws-amplify/amplify-android#2604]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin - Coroutines

### Amplify Categories

Authentication

### Gradle script dependencies

<details>

```groovy
// Put output below this line
def amplify_version = "2.13.2"
implementation "com.amplifyframework:core-kotlin:$amplify_version"
implementation "com.amplifyframework:aws-api:$amplify_version"
implementation "com.amplifyframework:aws-auth-cognito:$amplify_version"
implementation "com.amplifyframework:aws-storage-s3:$amplify_version"

```

</details>

### Environment information

<details>

```
# Put output below this line

------------------------------------------------------------
Gradle 7.5
------------------------------------------------------------

Build time:   2022-07-14 12:48:15 UTC
Revision:     c7db7b958189ad2b0c1472b6fe663e6d654a5103

Kotlin:       1.6.21
Groovy:       3.0.10
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          17.0.7 (JetBrains s.r.o. 17.0.7+7-b964.1)
OS:           Mac OS X 14.0 aarch64

```

</details>

### Describe the bug

After calling `Amplify.Auth.signIn(email)` and transitioned to the `SigningIn` state where the user is supposed to enter a token, I am not able to cancel the sign in flow so I can re-authenticate with other credentials.

I always get `Auth state is an invalid state, cannot process the request.`. Is there a way to cancel the current sign in flow so I can restart?

### Code Snippet

```java
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by gpanshu** [comment]
Hi @bayo-code can you add the debug logs and provide the logs for states. You can add `Amplify.addPlugin(AndroidLoggingPlugin(LogLevel.VERBOSE))` as one of your plugins for debugging locally as the first plugin in your application class. Please also advise what sign in method you are using?

**Comment by gpanshu** [comment]
Closing due to inactivity. Please feel free to comment here or create a new issue if you are still facing an issue.

**Comment by tylerjroach** [comment]
@bayo-code I apologize that this ticket was ever closed. This was a legitimate bug report. This issue has been fixed with Amplify v2.25.0.
